kiwi/filesystem/base.py

```diff
@@ -150,7 +163,6 @@ def sync_data(self, exclude=None):
             options=['-a', '-H', '-X', '-A', '--one-file-system'],
             exclude=exclude
         )
-        self.filesystem_mount.umount()
 
```

Comment:
umount takes place on deletion through the destructor when the object looses its scope. The overall problem with a chrooted script hook into the fully mounted disk space is that sync_data explicitly umounts after invocation. However all those classes were designed to cleanup when they loose its scope.

I have tested the order of destruction and that works now again as it was designed originally. The cleanup of the custom boot image code (boot/image/builtin_kiwi) has still an issue I'm currently working on. All the rest is clean... more clean than before because I realized there were object references that broke the correct destructor order

I know it's hard to read therefore better reference the log from the integration tests and watch out for messages like "Cleaning up ..." and their order

Reply:
> umount takes place on deletion through the destructor when the object looses its scope.

Oh damn, sorry I missed that. I looked at the code but got confused by the `builder/filesystem.py` class. Yes now I see the that the destructor of this class already includes the umount.